### PR TITLE
Use LT_INIT instead of obsolete AM_PROG_LIBTOOL

### PR DESCRIPTION
**Pull request by evpobr**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,2 @@
 language: c
-script:
--  - ./bootstrap.sh
--  - configure
--  - make distcheck
+script: ./autogen.sh && configure && make distcheck
```
Comment (rillian) [inline .travis.yml]: Why is it better to have everything on one line?

Comment (evpobr) [inline .travis.yml]: I guess it is the same.

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
-  - make distcheck
+- ./autogen.sh
+- ./configure
+- make distcheck
+os:
+- linux
+- osx
+compiler:
+- gcc
+- clang
```
Comment (rillian) [inline .travis.yml]: FWIW, I used a single entry here on purpose. I don't expect any platform differences between osx and linux or between clang and gcc for this library (msvc and warnings from newer compiler versions are a different story...) so it didn't seem worthwhile to spend travis resources and slow down patch feedback.

Comment (evpobr) [inline .travis.yml]: If you don't expect doesn't mean it doesn't exist.

Clang may give useful warnings in build log, different from gcc.

And under osx make install fails, no xz there.

**Comment by tmatth** [comment]
I've cherry-picked the autotools (i.e. LT_INIT) only changes which pass Travis, I will split out the Travis changes into a separate patch unless @rillian objects.

**Comment by rillian** [comment]
I don't object.

**Comment by rillian** [comment]
Closing this as obsolete. The `LT_INIT` change was merged as 572618e3f50f214a875d72b631115b67450fa939.

**Pull request closed without merging**